packages/data-transformer/src/transformers/data-transformer.builder.ts

```diff
@@ -0,0 +1,156 @@
+import {injectable} from "tsyringe";
+import {DataNormalizerUniqueKey} from "../types/data-normalizer-unique-key.type";
+import {DataNormalizerAlreadyAdded} from "../errors/data-normalizer-already-added.error";
+import {DataTransformerProperty} from "./data-transformer.property";
+import {moduleScoped} from "@pristine-ts/common";
+import {DataTransformerModuleKeyname} from "../data-transformer.module.keyname";
+import {DataTransformerInterceptorUniqueKeyType} from "../types/data-transformer-interceptor-unique-key.type";
+import {
+    DataAfterRowTransformerInterceptorAlreadyAddedError
+} from "../errors/data-after-row-transformer-interceptor-already-added.error";
+import {
+    DataBeforeRowTransformerInterceptorAlreadyAddedError
+} from "../errors/data-before-row-transformer-interceptor-already-added.error";
+
+@moduleScoped(DataTransformerModuleKeyname)
+@injectable()
+export class DataTransformerBuilder {
+    public normalizers: { key: DataNormalizerUniqueKey, options: any}[] = [];
+    public beforeRowTransformInterceptors: { key: DataTransformerInterceptorUniqueKeyType, options: any}[] = [];
+    public afterRowTransformInterceptors: { key: DataTransformerInterceptorUniqueKeyType, options: any}[] = [];
+
+    public properties: {[sourceProperty in string]: DataTransformerProperty} = {}
+
+    public addNormalizer(normalizerUniqueKey: DataNormalizerUniqueKey, options?: any): DataTransformerBuilder {
+        if(this.hasNormalizer(normalizerUniqueKey)) {
```

Comment:
This will throw if a normalizer already exists. What happens if you want the same normalizer on different properties ?

Reply:
Ohh never mind it's added on the property